README.md

```diff
@@ -23,7 +23,17 @@ Currently, the following encoding tokens are supported, using the first acceptab
 4. `*` (no preference — `fastify-compress` will use `gzip`)
 5. `identity` (no compression)
 
-If the `accept-encoding` header is missing or contains no supported encodings, it will not compress the payload.
+If an unsupported encoding is received or if the `'accept-encoding'` header is missing, it will not compress the payload. If an unsupported encoding is received, a custom error response can be sent in place of the uncompressed payload by setting the `onUnsupportedEncoding(encoding, reply)` option to be a function that can modify the reply and return a `string | Buffer | Stream | Error` payload.
+```javascript
+fastify.register(
+  require('fastify-compress'),
+  {
+    onUnsupportedEncoding: (encoding, reply) => {
```

Comment:
I would also pass the `request` object as an argument, you can access it with `reply.request`.

Reply:
I was thinking a user could access the request if needed with `reply.request` in the callback. Do you think it is worth explicitly providing it?

Reply:
Thoughts?

Reply:
It is easier for the user to take actions, like read the payload in request and log it

Reply:
Ok! I'll do this.